oauth2_http/java/com/google/auth/oauth2/ExternalAccountCredentials.java

```diff
@@ -208,6 +210,27 @@ private ImpersonatedCredentials initializeImpersonatedCredentials() {
         .build();
   }
 
+  @Override
```

Comment:
Note to reviewer: before this PR, the blocking version getRequestMetadata() was the primary codepath, so ExternalAccountCredentials only had to override that method. This subclass expected that the async version would just call the sync version. Which I think is a poor assumption since its an implementation detail. I corrected the behavior by overriding both.

Arguably a better approach is to make both getRequestMetadata final and have subclasses use  asyncFetch() as the source of truth, but I didn't want to leak guava to the surface of this library